src/udev/udev-builtin-hwdb.c

```diff
@@ -99,14 +99,25 @@ static int udev_builtin_hwdb_search(sd_device *dev, sd_device *srcdev,
                         last = true;
                 }
 
-                if (!modalias)
-                        goto next;
+                if (streq(dsubsys, "firewire")) {
+                        /* The node device has character device while the unit device doesn't. */
+                        if (sd_device_get_devname(d, NULL) >= 0) {
```

Comment:
Not a matter, but the first `{` can be dropped, also, you can merge the two `if` conditions. I.e.,
```
if (streq(dsubsys, "firewire") && sd_device_get_devname(d, NULL) >= 0) {
    ...;
}
```

Reply:
> Not a matter, but the first { can be dropped, also, you can merge the two if conditions.

Indeed. I'll pushed revised version later.